kolibri/core/webpack/hooks.py

```diff
@@ -0,0 +1,290 @@
+        return bundle
+
+    @property
+    @hooks.registered_method
+    def webpack_bundle_data(self):
+        """
+        This is the main interface to the NPM Webpack building util. It is
+        used by the webpack_json management command. Inheritors may wish to
+        customize this.
+
+        :returns: A dict with information expected by webpack parsing process.
+        """
+        return {
+            "name": self.unique_slug,
+            "src_file": self.src_file,
+            "static_dir": self.static_dir,
+            "stats_file": self.stats_file,
+            "events": {},
+            "once": {},
+        }
+
+    @property
+    def build_path(self):
+        """
+        An auto-generated path to where the build-time files are stored,
+        containing information about the built bundles.
+        """
+        return os.path.join(
+            os.path.abspath(os.path.dirname(__name__)),
+            self._module_file_path,
+            "build"
+        )
+
+    @property
+    def stats_file(self):
+        """
+        TODO: Do we want to rely on a generated stats file? It will have to be
+        read for every bundle, every time stuff is loaded.
+
+        An auto-generated path to where the build-time files are stored,
+        containing information about the built bundles.
+        """
+        return os.path.join(
+            self.build_path,
+            "{plugin}_stats.json".format(plugin=self.unique_slug)
+        )
+
+    @property
+    def _module_file_path(self):
+        """
+        Returns the path of the class inheriting this classmethod.
+        """
+        return os.path.join(*self.__module__.split(".")[:-1])
+
+    def render_to_html(self, extension=None):
+        """
+        This function tags a bundle of file chunks and generates the appropriate
+        script tags for them, be they JS or CSS files.
+
+        :param bundle_data: The data returned from
+        :return: HTML of script tags for insertion into a page.
+        """
+        tags = []
+        js_tag = '<script type="text/javascript" src="{url}"></script>'
+        css_tag = '<link type="text/css" href="{url}" rel="stylesheet"/>'
+        for chunk in self.bundle_filtered(extension=extension):
+            if chunk['name'].endswith('.js'):
+                tags.append(js_tag.format(url=render_as_url(chunk)))
+            elif chunk['name'].endswith('.css'):
+                tags.append(css_tag.format(url=render_as_url(chunk)))
+        return mark_safe('\n'.join(tags))
+
+
+class FrontEndAssetHook(WebpackBundleHook):
+    """
+    An abstract class for all assets destined for the default front-end. You
+    probably want to use FrontEndSyncHook or FrontEndASyncHook to be explicit.
+
+    Everything inheriting from this hook will automatically be included in the
+    front-end.
+    """
+
+    class Meta:
+        abstract = True
+
+
+class FrontEndSyncHook(FrontEndAssetHook):
```

Comment:
Essentially they are just unused building blocks, and a mock-up where I *think* I implemented the previous hooks... please feel welcome to change the structure to reflect how frontend assets are grouped from here on :)